ir_api/core/utility.py

```diff
@@ -0,0 +1,19 @@
+"""Collection of utility functions"""
+import functools
+from typing import Callable, Any, TypeVar, cast
+
+from ir_api.core.exceptions import UnsafePathError
+
+FuncT = TypeVar("FuncT", bound=Callable[[str], Any])
+
+
+def forbid_path_characters(func: FuncT) -> FuncT:
+    """Decorator that prevents path characters {/, ., \\} from a functions args by raising UnsafePathError"""
+
+    @functools.wraps(func)
+    def wrapper(arg: str) -> Any:
+        if any(char in arg for char in (".", "/", "\\")):
```

Comment:
```suggestion
        if not arg.isalnum():
```
Would it make sense to use this, not necessarily here but do a generic alphanumeric check, not for the path but for the instrument that is passed?

Reply:
I thought about this as I was doing it, I didn't do that because it felt likely that this could come up again in the future in places outside of instrument. I like the alnum change thought. It should have occured to me 🤦

Reply:
Also the check doesn't happen here, it happens where the function is decoracted in `acquision.py`